CHANGES.md

```diff
@@ -1,5 +1,33 @@
 # Change log for kotlinx.coroutines
 
+## Version 0.27.0 
+
+* **[Major]** Public API revision. All public API was reviewed and marked as preparation to `1.0` release:
+   1. `@Deprecated` API. All API marked as deprecated will be removed in 1.0 release without replacement.
+   2. `@ExperimentalCoroutinesApi` API. This API is experimental and may change in the future, but migration mechanisms will be provided. Signature, binary compatibility and semantics can be changed.
+   3. `@InternalCoroutinesApi`. This API is intended to be used **only** from within `kotlinx.coroutines`. It can and will be changed, broken 
+       and removed in the future releases without any warnings and migration aids. If you find yourself using this API, it is better to report
+       your use-case to Github issues, so decent, stable and well-tested alternative can be provided.
+   4. `@ObsoleteCoroutinesApi`. This API has serious known flaws and will be replaced with a better alternative in the nearest releases.
+   5. Regular public API. This API is proven to be stable and is not going to be changed. If at some point it will be discovered that such API
+      has unfixable design flaws, it will be gradually deprecated with proper replacement and migration aid, but won't be removed for at least a year.
+* **[Major]** Job state machine is reworked. It includes various performance improvements, fixes in 
+data-races which could appear in a rare circumstances and consolidation of cancellation and exception handling.
+Visible consequences of include more robust exception handling for large coroutines hierarchies and for different kinds of `CancellationException`, transparent parallel decomposition and consistent view of coroutines hierarchy in terms of its state (see #220 and #585).
+* NIO, Quasar and Rx1 integration modules are removed with no replacement.
```

Comment:
It is good to mention issue #s here for references of those who want to discuss it.